File: 6-Patron/php/Bendicion_BotasRecorrer.php
<?php
include './Botas.php';
class Bendicion_BotasRecorrer extends Botas{
	private $habilidad;

	public function __construct($habilidad){
    	$this->habilidad = $habilidad;
	}

	public function getHabilidad(){
		return $this->habilidad;	
	}

	public function setHabilidad($habilidad){
		$this->habilidad = $habilidad;
	}


	public function verBotas(){
		echo 'Velocidad: ' . $this->velocidad . 'Habilidad: ' . $this->habilidad;

	}

	public function activarHabilidadRetribucion(){
		$this->habilidad = 'Retribucion del angel caido';
		echo 'La habilidad: ' . $habilidad . ' ha sido activada.';
	}

}
?>

File: 6-Patron/Java/Bendicion_BotasRecorrer.java
package patron6;

public class Bendicion_BotasRecorrer extends Botas{
    private String Habilidad;

    
    public Bendicion_BotasRecorrer(String Habilidad) {
        this.Habilidad = Habilidad;
    }

    public String getHabilidad() {
        return Habilidad;
    }

    public void setHabilidad(String Habilidad) {
        this.Habilidad = Habilidad;
    }
    
    public void activarHabilidadFuria(){
        this.Habilidad = "Furia divina";
    }
    @Override
    public void verBotas() {
        System.out.println("Habilidad: "+this.Habilidad + "Velocidad: " );
    }
    
}
